kedro/framework/hooks/manager.py

```diff
@@ -48,6 +48,8 @@ def _register_hooks(hook_manager: PluginManager, hooks: Iterable[Any]) -> None:
         if not hook_manager.is_registered(hooks_collection):
             hook_manager.register(hooks_collection)
 
+    hook_manager.check_pending() # Validate hook_impl respect hook_spec
```

Comment:
What's this for?

Reply:
Ignore it for now, I have no idea maybe I merged from other branches accidentally.